### ISSUE TITLE
Using custom loss with VisionClassifierTrainer 

### ISSUE THREAD

**Issue by guneetsk99** [qanastek/HugsVision#40]
I wanted to use a custom loss function with VIT how should I proceed since in the VisionClassifierTrainer there is no use of loss

**Comment by qanastek** [comment]
Hi,

Since the HugsVision `VisionClassifierTrainer` is based on the HuggingFace transformers library and especially on their `Trainer`, we cannot tweak the loss function.

![image](https://user-images.githubusercontent.com/19389475/167636019-459ce29f-fcdc-443d-b426-6ea088849dc8.png)

**Comment by guneetsk99** [comment]
Thanks @qanastek for the prompt response
If possible can you suggest how I can use your models/ directory since there I can change the crossentropy loss to for eg KLDivLoss for testing

**Comment by qanastek** [comment]
HugsVision isn't really suited for reasearch. But you can directly rewrite the Transformers class [ViTForImageClassification](https://github.com/huggingface/transformers/blob/31ec2cb2badfbdd4c1ac9c6c9b8a74e974984206/src/transformers/models/vit/modeling_vit.py#L789) for the `single_label_classification` task.

In my opinion, it's the simplest way to do. Clone the [Transformers repository](https://github.com/huggingface/transformers/tree/v4.18.0), install it locally using `pip install --editable .` and modify the loss in the class.
